catapult-sdk/src/plugins/lockSecret.js

```diff
@@ -35,23 +35,25 @@ const lockSecretPlugin = {
 			lock: { type: ModelType.object, schemaName: 'secretLockInfo.lock' }
 		});
 		builder.addSchema('secretLockInfo.lock', {
-			account: ModelType.binary,
-			accountAddress: ModelType.binary,
+			senderPublicKey: ModelType.binary,
+			senderAddress: ModelType.binary,
 			mosaicId: ModelType.uint64,
 			height: ModelType.uint64,
 			secret: ModelType.binary,
-			recipient: ModelType.binary
+			recipientAddress: ModelType.binary
 		});
 
 		builder.addTransactionSupport(EntityType.secretLock, {
 			mosaicId: ModelType.uint64,
 			duration: ModelType.uint64,
+			amount: ModelType.uint64,
```

Comment:
why `amount`? it's not in catbuffer: https://github.com/nemtech/catbuffer/blob/master/schemas/lock_secret/secret_lock.cats

Reply:
`Amount` is in mongoDB. but I think you are right, it's removed.

Reply:
need to add amount to codecs (below) - note that ordering matters since codecs serialize and deserialize binary transaction data (must be put after `mosaic`)

Reply:
Fixed